transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -120,10 +120,31 @@ protected void run() {
 
         // Lets add the eventfd related events before starting to do any real work.
         submissionQueue.addPollIn(eventfd.intValue());
-        submissionQueue.submit();
 
         for (;;) {
             logger.trace("Run IOUringEventLoop {}", this.toString());
+            // avoid blocking for as long as possible - loop until two consecutive "empty" results
+            boolean maybeMoreTasks = true;
+            boolean maybeMoreIo;
+            do {
+                maybeMoreIo = completionQueue.process(this) != 0;
+                if (!maybeMoreIo & !maybeMoreTasks) {
+                    break; // no need to check tasks again here
+                }
+                maybeMoreTasks = runAllTasks();
+                try {
+                    if (isShuttingDown()) {
+                        closeAll();
+                        if (confirmShutdown()) {
+                            return;
+                        }
+                    }
+                } catch (Throwable t) {
+                    logger.info("Exception error: {}", t);
```

Comment:
I think the exception should say something about "Caught exception when shutting down event loop" or something. This part, including the error handling, could also be extracted to a separate method, so it looks like "if (blabla()) { return; }`

Reply:
@chrisvest agree this needs more changes but I had just moved this part from lower down. I think there's a fair amount still to be done for error handling in general so maybe best for another PR(s)?

Reply:
Sounds good.